src/plugins/action-queue/cleaners.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+
+//
+// Action Operations
+//
+import { type Cleaner, asArray, asEither, asNumber, asObject, asOptional, asString, asValue } from 'cleaners'
+
+import { type ActionEffect, type ActionOp, type ActionProgram, type ActionProgramState } from './types'
+
+const asError = (raw: any) => {
```

Comment:
Error object will not round-trip through `JSON.parse`. Maybe we need to be explicit and grab `name` and `message` from the error object. So `{ name: 'TypeError', message: 'Expected a string at .nativeAmount' }` becomes what we write / read. If you use `.withRest`, I think it will also preserve the extra properties you find on core errors like `SwapAboveLimitError`.

You may need `asCodec` to make this round-trip to an actual `Error` instance. Updating the `.name` property requires `Object.defineProperty`, as it is not writable by default.

Reply:
Let's go with

```
{
  type: 'done',
  error?: {
    message: string
  }
}
```

Reply:
Instead of changing the runtime type and cleaning it after catching the error, I implemented a serialization type for the error in the `asError` cleaner so we only need to clean/unclean the type during disk read/write.